### PR TITLE
[ADD] payment_worldline: integrate with Worldline Direct

### PR DESCRIPTION
**Pull request by Demesmaeker**
Ogone & SIPS are deprecated by Worldline. Their focus is on the Worldline Direct API. Worldline plans to centralize all its businesses (Ogone, SIPS...) in the Worldline Direct API. Worldline Direct shares the same engine as Ogone, which makes the migration from Ogone to Direct easy for customers.

task-3499275

See also:
- https://github.com/odoo/upgrade/pull/6498

### CODE REVIEW

File: addons/payment/data/payment_method_data.xml [hunk 1]
```diff
@@ -877,44 +946,44 @@
...
-                         ref('base.GBP'),
+                         ref('base.IDR'),
                      ])]"
         />
     </record>
 
-     <record id="payment_method_cimb_niaga" model="payment.method">
-        <field name="name">CIMB Niaga</field>
-        <field name="code">cimb_niaga</field>
+    <record id="payment_method_clearpay" model="payment.method">
```
Comment (Pirols) [inline addons/payment/data/payment_method_data.xml:970]: From the upgrade PR. Are you just swapping the order of clearpay and cimb niaga or there is an actual change happening here?

Comment (Demesmaeker) [reply addons/payment/data/payment_method_data.xml:970]: Just swapping, they were all in alphabetical order except that one, it bothered me... a lot. (I'm weak xD)

File: addons/payment_worldline/tests/test_worldline.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+from odoo.tests import tagged
+from odoo.tools import mute_logger
+
+from odoo.addons.payment.tests.http_common import PaymentHttpCommon
+from odoo.addons.payment_worldline.controllers.main import WorldlineController
+from odoo.addons.payment_worldline.tests.common import WorldlineCommon
+
+
+@tagged('post_install', '-at_install')
+class WorldlineTest(WorldlineCommon, PaymentHttpCommon):
```
Comment (AntoineVDV) [inline addons/payment_worldline/tests/test_worldline.py:20]: Let's keep it as-is for now, but it'd be great to split the tests to easy their understanding and maintenance. See [task 3054481](https://www.odoo.com/odoo/action-1592/new/4106/tasks/3054481).

**Review by AntoineVDV — APPROVED** [review]
Clean work! :ok_hand: 

There were a few things that were left out, see [the diff](https://github.com/odoo/odoo/compare/5904edeea541aab9f61a68c4975bc2ac334f583d..52f401eb16437af3664563423d1c5b491888a21e) to find them. Other than that, the submitted version was working fine and required only small improvements.

@robodoo r+ rebase-ff

@robodoo override=ci/style

@robodoo override=ci/security
Reason:
- `csrf=False`: origin of the request is checked by querying a fixed HTTPS API URL.
- `hmac.new()`/`hashlib.sha256`: used to compute a signature based on another secret than the db secret.
- `hmac.compare_digest()`: used to compare two signatures.
- `requests.post`: request to a fixed URL with limited timeout.

Link to ci/security build (master): https://runbot.odoo.com/runbot/batch/1621091/build/68361124

**Comment by AntoineVDV** [comment]
@robodoo r+

**Comment by AntoineVDV** [comment]
@robodoo r+

File: addons/payment/data/payment_method_data.xml [hunk 3]
```diff
@@ -924,13 +993,14 @@
         <field name="code">cofidis</field>
         <field name="sequence">1000</field>
         <field name="active">False</field>
-        <field name="image" type="base64" file="payment/static/img/cofidis.png"/>
+        <field name="image" type="base64" file="payment/static/img/clearpay.png"/>
```
Comment (MiquelRForgeFlow) [inline addons/payment/data/payment_method_data.xml:996]: Why change image? It doesn't seem right.

**Pull request closed without merging**